toolchain/check/testdata/function/declaration/no_prelude/name_poisoning.carbon

```diff
@@ -0,0 +1,237 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/function/declaration/no_prelude/name_poisoning.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/function/declaration/no_prelude/name_poisoning.carbon
+
+// --- no_poison.carbon
+
+library "[[@TEST_NAME]]";
+
+namespace N;
+class C {};
+class N.C {}
+fn N.F1(x: C);
+
+fn N.F2(x: C) {N.F1(x);}
+
+// --- poison_without_usage.carbon
+
+library "[[@TEST_NAME]]";
+
+namespace N;
+class C {};
+fn N.F1(x: C);
+
+// TODO: Should fail here.
+class N.C {}
+
+// --- fail_poison_with_usage.carbon
+
+library "[[@TEST_NAME]]";
+
+namespace N;
+class C {};
+fn N.F1(x: C);
+
+// TODO: Should fail here.
+class N.C {}
+// CHECK:STDERR: fail_poison_with_usage.carbon:[[@LINE+6]]:21: error: package `Core` implicitly referenced here, but not found [CoreNotFound]
```

Comment:
I'm having trouble understanding why this error is occurring. Is it an intended part of the test?

Reply:
Yes, this error is misleading and I hope to see it removed when name poisoning is properly handled.
Since both `C` and `N.C` classes are defined, the type of `x` in `N.F2()` is `N.C` while `N.F1()` parameter is of type `C`, so a conversion is necessary here.

Reply:
FWIW, I've implemented some version of name poisoning and indeed this error is gone.

Reply:
OK, then can you add a TODO comment explaining that it shouldn't fail here?

Reply:
Done